storage/feed/handler.go

```diff
@@ -192,12 +192,10 @@ func (h *Handler) Lookup(ctx context.Context, query *Query) (*cacheEntry, error)
 		ctx, cancel := context.WithTimeout(ctx, defaultRetrieveTimeout)
 		defer cancel()
 
-		ch, err := h.chunkStore.Get(ctx, chunk.ModeGetLookup, id.Addr())
+		r := storage.NewRequest(id.Addr())
+		ch, err := h.chunkStore.Get(ctx, chunk.ModeGetLookup, r)
 		if err != nil {
-			if err == context.DeadlineExceeded { // chunk not found
```

Comment:
`context.DeadlineExceeded` is not the only error that signifies that a chunk is not found. Currently the NetStore returns a few different errors which explain in detail why a chunk cannot be found, and we don't have them as error values. This is something we might want to refactor in the future.

Reply:
cc @jpeletier

Reply:
@nonsense This change is affecting the behavior. If a context is cancelled, the lookup algorithm needs to get a context cancelled error (`context.Canceled`) and you'd be returning `nil` all the time, regardless of the error.

I'd leave this code as is until the NetStore offers some explicit error values I can check.

Reply:
I see this checking for `context.DeadlineExceeded` and not for `context.Canceled`... so could you explain a bit further the behaviour you rely on here?

Reply:
The lookup algorithm checks the value returned by this function for `context.Canceled`: `feed/lookup/algorithm_longearth.go`.

The change you are proposing makes this function to never return an error. We need to know if the context was canceled to see if this is an "intentional" error or something happened.

Reply:
Thanks for the pointer. Refactored the code, and exported an error value, which you would also get when `chunk is not found` - `ErrNoSuitablePeer` - this happens when your node knows that it has no more peers left to ask for a given chunk and quits searching.

I also fixed the code to correctly return the `Canceled` and `DeadlineExceeded` errors (although I don't believe we have tests covering that right now)